arcade/tilemap/tilemap.py

```diff
@@ -411,6 +412,7 @@ def _create_sprite_from_tile(
 
         # --- Step 1, Find a reference to an image this is going to be based off of
         map_source = self.tiled_map.map_file
+        assert map_source
```

Comment:
I see the idea, but asserts should not be used in production code,
it is seen as an Anti-Pattern.

please use an if raise block.

Reply:
I get the argumentation, but if map_file can never be `None`, why do we have to fix the misleading optional here and not where it is created?

`self.tiled_map.map_file` should not return an Optional in that case.

Reply:
I'm not sure, maybe @Cleptomania can weight in here.

The optional comes from here:
https://github.com/pythonarcade/pytiled_parser/blob/47297a72c700f9b8cd558ef26420bc742b84f7f2/pytiled_parser/tiled_map.py#L72

In our case, `_create_sprite_from_tile` is called by `_process_object_layer` and `_process_tile_layer`, which are called by `_process_layer`, which is only called by `__init__`.  `__init__` accepts either a path to a map, or a pre-parsed map.
https://github.com/pythonarcade/arcade/blob/development/arcade/tilemap/tilemap.py#L108-L118

Reply:
I asked @Cleptomania about this on Discord: https://discord.com/channels/458662222697070613/1105548182768865330/1106672539075362867

Based on that, I went with the `if TYPE_CHECKING: assert` and added comments about the guarantees that pytiled_parser makes despite the type hints.